<?xml version="1.0"?>
<launch>

	<!-- arg -->
	<arg name="gui" 		default="False" />
	<arg name="hand_frame_vision" 	value="/hand_root"/>
	<arg name="world_frame" 	value="/world"/>
        <arg name="hand_frame"		value="/hand"/>
	<!--arg name="record_folder"	value="$(find record)/saved/"/-->
	<arg name="path_sensor_model"   value="$(find objects)/meshes/plug/config/X.txt"/>
	<param name="use_gui" 		value="$(arg gui)"/>


	<arg name="socket_type"		default="table_sw_one"/>
	<arg name="table_urdf" value="'$(find peg_in_hole)/urdf/One/table_sw_one.urdf'" />
	<!-- table socket -->
	<include file="$(find peg_in_hole)/launch/table/$(arg socket_type).launch">
	  	<arg name="use_rviz" 			value="False"/>
		<arg name="origin"   			value="0 0 0"/>
		<arg name="orientation"			value="0 0 0 1"/>
		<arg name="load_saved"			value="False"/>	
		<arg name="use_vision_broadcaster" 	value="False"/>
		<arg name="use_broadcaster" 		value="True"/>
	</include> 


	<!-- peg handle -->
	<include file="$(find peg_handle)/launch/peg_handle.launch">
	  	<arg name="use_rviz" 			value="False"/>
		<arg name="origin"   			value="0.4 0 0"/>
		<arg name="orientation"			value="1 0 1 0"/>
		<arg name="load_saved"			value="False"/>		
		<arg name="use_vision_broadcaster" 	value="False"/>
		<arg name="use_broadcaster" 		value="True"/>
	</include>
	
	

	<!-- launch optitrack -->
	<!--include file="$(find optitrack)/launch/optitrack.launch" /-->

	<!-- feature wrapper  --><!-- launch-prefix="xterm -e gdb -args"-->
	<!--node name="peg_in_hole_wrapper" pkg="peg_in_hole" type="peg_vis_feature_node" args="-urdf $(arg table_urdf) -path_sensor_model $(arg path_sensor_model) -rate 100" output="screen"/-->


	
	<!--include file="$(find peg_in_hole)/launch/filter.launch"/-->


	<!-- rviz -->
	<node name="rviz" pkg="rviz" type="rviz"  args="-d $(find peg_in_hole)/launch/rviz/peg_in_hole.rviz"  required="true" />



</launch>
